honeybadger/contrib/aws_lambda.py

```diff
@@ -0,0 +1,118 @@
+import sys
+
+from honeybadger import honeybadger
+from honeybadger.plugins import Plugin
+from honeybadger.utils import filter_dict
+
+from threading import local
+
+_thread_locals = local()
+REQUEST_LOCAL_KEY = '__awslambda_current_request'
+
+def current_event():
+    """
+    Return current execution event for this thread.
+    """
+    return getattr(_thread_locals, REQUEST_LOCAL_KEY, None)
+
+def set_event(aws_event):
+    """
+    Set current execution event for this thread.
+    """
+
+    setattr(_thread_locals, REQUEST_LOCAL_KEY, aws_event)
+
+def clear_event():
+    """
+    Clears execution event for this thread.
+    """
+    if hasattr(_thread_locals, REQUEST_LOCAL_KEY):
+        setattr(_thread_locals, REQUEST_LOCAL_KEY, None)
+
+def reraise(tp, value, tb=None):
+    """
+    Re-raises a caught error
+    """
+    assert value is not None
+    if value.__traceback__ is not tb:
+        raise value.with_traceback(tb)
+    raise value
+    
+def get_lambda_bootstrap():
+    """
+    Get AWS Lambda bootstrap module
+    """
+
+    if "bootstrap" in sys.modules:
+        return sys.modules["bootstrap"]
+    elif "__main__" in sys.modules:
+        if hasattr(sys.modules["__main__"], "bootstrap"):
+            return sys.modules["__main__"].bootstrap
+        return sys.modules["__main__"]
+    else:
+        return None
+
+def _wrap_lambda_handler(handler):
+
+    def wrapped_handler(aws_event, aws_context, *args, **kwargs):
+        set_event(aws_event)
+
+        honeybadger.begin_request(aws_event)
+        try:
+            return handler(aws_event, aws_context, *args, **kwargs)
+        except Exception as e:
+            honeybadger.notify(e)
+            exc_info = sys.exc_info()
+            clear_event()
+            honeybadger.reset_context()
+
+            #Rerase exception to proceed with normal aws error handling
+            reraise(*exc_info)
+
+    return wrapped_handler
+
+
+class AWSLambdaPlugin(Plugin):
+
+    def __init__(self):
+        super(AWSLambdaPlugin, self).__init__('AWSLambda')
+        lambda_bootstrap = get_lambda_bootstrap()
+        self.initialize_request_handler(lambda_bootstrap)
+        
+    def supports(self, config, context):
+        return config.is_aws_lambda_environment
+
+    def generate_payload(self, config, context):
+        """
+        Generate payload by checking the lambda's
+        request event
+        """
+        payload = {
+            "event": current_event(),
```

Comment:
@Kelvin4664 What is the value of "event" here, and where is it currently used? Wouldn't this be added to the `notice["request"]` payload (which doesn't support an event key)?

Reply:
Ah!.. my bad. i thought we can pass arbitrary keys.
However i believe it is important to keep a log of the event that triggers an error. Where do you suggest we pass that from?

Reply:
@Kelvin4664 I'd say either in `notice["request"]["context"]`, or in `notice["details"]["Lambda Details"]`. If it would be useful to split up the event into its own section, we can also have several details sections, so we could have something like: `notice["details"]["Lambda Environment"]` (for the current Lambda Details) and `notice["details"]["Lambda Event"]`.

@stympy @rabidpraxis as a Lambda+Honeybadger user, which do you prefer?

Reply:
My vote would be to put it in the "Lambda Details" section.

Reply:
If we're talking about the hash that arrives as the first parameter of the function, then it makes sense to put that into params.

Reply:
Ah, I didn't think of params. @Kelvin4664 let's put the event in `notice["request"]["params"]`. Actually, I'll take a shot at fixing that, since I think it's just a rename. :)

Reply:
I believe this should do it: https://github.com/honeybadger-io/honeybadger-python/pull/67/commits/48cb68b1735611abf06697faf6609f68ced5c603